pkg/controller/stackconfigpolicy/elasticsearch_config_settings.go

```diff
@@ -119,3 +121,23 @@ func getElasticsearchConfigAndMountsHash(elasticsearchConfig *commonv1.Config, s
 	}
 	return hash.HashObject(secretMounts)
 }
+
+// elasticsearchConfigAndSecretMountsApplied checks if the elasticsearch config and secret mounts from the stack config policy have been applied to the Elasticsearch cluster.
+func elasticsearchConfigAndSecretMountsApplied(ctx context.Context, c k8s.Client, policy policyv1alpha1.StackConfigPolicy, es esv1.Elasticsearch) (bool, error) {
+	// Get Pods for the given Elasticsearch
+	podList := corev1.PodList{}
+	if err := c.List(ctx, &podList, client.MatchingLabels{
```

Comment:
Could it be an issue if there is 2 ES in 2 namespaces and only one managed by a SCP?
In any case, it's better to reduce the spectrum of a list to the namespace when we can.
​
```suggestion
	if err := c.List(ctx, &podList, client.InNamespace(es.Namespace), client.MatchingLabels{
```